system/utils.py

```diff
@@ -79,12 +85,63 @@ def random_seed_and_json():
     return trustee_did, trustee_vk
 
 
+# TODO why we need that async ???
 async def payment_initializer(library_name, initializer_name):
     library = CDLL(library_name)
     init = getattr(library, initializer_name)
     init()
 
 
+async def eventually(awaited_func,
+                     *args,
+                     retry_wait: float=0.1,
+                     timeout: float=5,
+                     acceptableExceptions=None,
+                     verbose=True,
+                     **kwargs):
+    if not timeout > 0:
+        raise ValueError("'timeout' is {}".format(timeout))
+    if acceptableExceptions and not isinstance(acceptableExceptions, Iterable):
+        acceptableExceptions = [acceptableExceptions]
+
+    start = time.perf_counter()
+
+    fname = awaited_func.__name__
+    while True:
+        remain = 0
+        try:
+            remain = start + timeout - time.perf_counter()
+            if remain < 0:
+                # this provides a convenient breakpoint for a debugger
+                logger.debug("{} last try...".format(fname))
+            # noinspection PyCallingNonCallable
+            res = awaited_func(*args, **kwargs)
+
+            if isawaitable(res):
+                result = await res
+            else:
+                result = res
+
+            if verbose:
+                logger.debug("{} succeeded with {:.2f} seconds to spare".
+                             format(fname, remain))
+            return result
+        except Exception as ex:
+            if acceptableExceptions and type(ex) not in acceptableExceptions:
+                raise
+            if remain >= 0:
+                sleep_dur = retry_wait
+                if verbose:
+                    logger.debug("{} not succeeded yet, {:.2f} seconds "
+                                 "remaining..., will sleep for {}".format(fname, remain, sleep_dur))
+                await asyncio.sleep(sleep_dur)
+            else:
+                logger.error("{} failed; not trying any more because {} "
+                             "seconds have passed; args were {}".
+                             format(fname, timeout, args))
+                raise ex
```

Comment:
It would be nice to log exception as well - sometimes tracebacks get eaten by pytest